### ISSUE TITLE
Add anonymous upload

### ISSUE THREAD

**Issue by luisjira** [LycheeOrg/Lychee#1148]
Give the ability to create an upload link to an album, from which anyone with the link can upload but not delete content.
This setting could potentially even be independent from the album being public.

This could be useful for sharing pictures in a group of people.

**Comment by Vinpasso** [comment]
+1 This would be really useful. The workaround of creating a user and sharing the password is a little bit cumbersome

**Comment by matthijskooijman** [comment]
I would also be interested in this. Ideally, it would also be impossible for different anonymous uploaders to see the pictures uploaded by others (so we can apply a moderation to the pictures before sharing them with others) and it would be possible (or probably even required) for an uploader to create a subalbum and upload in there, to keep pictures uploaded by different people separated.

**Comment by ildyria** [comment]
> Ideally, it would also be impossible for different anonymous uploaders to see the pictures uploaded by others (so we can apply a moderation to the pictures before sharing them with others) and it would be possible (or probably even required) for an uploader to create a subalbum and upload in there, to keep pictures uploaded by different people separated.

That is not possible you cannot have at the same time anonymity and user tracking. Unless we add a user fingerprinting solution in Lychee this seems a bit too far.

It is either you see every anonymous pictures uploaded or you see none...

**Comment by matthijskooijman** [comment]
> That is not possible you cannot have at the same time anonymity and user tracking. Unless we add a user fingerprinting solution in Lychee this seems a bit too far.

Something simpler like a (session) cookie would be fine here, I think? That means an (anonymous) user can work with their own files as long as the cookie stays active. They will loose access if they clear cookies (or restart the browser if using a session cookie), switch browsers, etc. but for the "anonymous upload" usecase that would seem acceptable (you just need to track a user long enough for them to upload their files and maybe have them doublecheck that everything was uploaded correctly directly afterwards).

**Comment by ildyria** [comment]
I would need to check but I am not sure we are opening a session when a user is not logged in.
Also that would mean that we would have to keep that data temporarily in the database too... 🤔

**Comment by Vinpasso** [comment]
For my use-case it would be perfectly sufficient if it would be possible to configure that every user with the link to the album has write access (perhaps via a revocable sharing link?)


> and it would be possible (or probably even required) for an uploader to create a subalbum and upload in there, to keep pictures uploaded by different people separated

If finer access control is needed for anonymous user uploads, then the creation of subalbums could also work for access control, as these could have their own (automatically generated) read/write links which the anonymous user could either forget at the end of the session or save somewhere for later editing. 
Then it would also theoretically be possible to configure whether the newly created subalbums should be immediately visible from the parent album or whether they should start invisible (to implement moderation by an album owner).

While the latter approach feels more versatile to me, I think this is also a little more complex to implement. Perhaps being able to just set permissions for the anonymous user on the album is simpler to start?

**Comment by matthijskooijman** [comment]
> If finer access control is needed for anonymous user uploads, then the creation of subalbums could also work for access control, as these could have their own (automatically generated) read/write links which the anonymous user could either forget at the end of the session or save somewhere for later editing.

Ah, that is a nice trick to prevent having to keep a session - just redirect the user to the newly created share link for the newly created subalbum and you link the subalbum to the uploading user in that way. Does mean that if they close the window and then reopen the original link, they will not have (write) access to the new album anymore, but that seems acceptable to me for the "dump your photos here" workflow that I have in mind.

**Comment by ildyria** [comment]
I agree that having a login link etc would be a more elegant alternative, unfortunately I lack the time to implement it.
LycheeOrg is always welcoming new team members. :)

After a quick dive in the code and some local tests, I figured out that it is quite easy to add the ability for unauthenticated users to upload pictures. Created a PR here: https://github.com/LycheeOrg/Lychee/pull/3069

Note that this upload is unlimited, so I would suggest to make the album hidden and give the link to interested parties instead of fully public...

**Comment by steilerDev** [comment]
@ildyria has this been released yet? I'm struggling to find the setting.

**Comment by ildyria** [comment]
Since [v6.3.5](https://lycheeorg.dev/docs/releases.html#v635) (March 19th)

- `SE` [♯3069](https://github.com/LycheeOrg/Lychee/issues/3069) : Allow anonymous users to upload by @ildyria.
  > This is only available for our Sponsors, this is also very niche option. The owner of the album can now decide whether anonymous users can upload to the album. There is no image validation, as a result, you are responsible for the content that is uploaded. Use at your own risk.
